site/public/js/server.js

```diff
@@ -16,8 +16,10 @@ window.addEventListener("load", function() {
  * @param {object} parts - Object representing URL parts to append to the URL
  * @returns {string} - Built up URL to use
  */
-function buildUrl(parts) {
-    var url = siteUrl;
+function buildUrl(parts, url="") {
+    if(url===""){
+        url = siteUrl;
+    }
```

Comment:
I'm not sure I see the point of this change? What is the use-case that we're worried that we'll need to build a URL to a different site?

Reply:
I think if that's the usecase, I'd then prefer just moving script.js to load at the end of the DOM and fire immediately then (when body would be available) than make this hack. Or examine why we're firing events to buildUrls before we've finished loading the initial DOM.